mito-ai/src/Extensions/AiChat/ChatMessage/ChatInput.tsx

```diff
@@ -33,42 +75,166 @@ const ChatInput: React.FC<ChatInputProps> = ({
         textarea.style.height = `${minHeight}px`;
     };
 
+    const handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        const value = event.target.value;
+        setInput(value);
+
+        const cursorPosition = event.target.selectionStart;
+        const textBeforeCursor = value.slice(0, cursorPosition);
+        const words = textBeforeCursor.split(/\s+/);
+        const currentWord = words[words.length - 1];
+
+        if (currentWord.startsWith("@")) {
+            const query = currentWord.slice(1);
+            const filtered = expandedVariables.filter((variable) =>
+                variable.variable_name.toLowerCase().includes(query.toLowerCase()) &&
+                variable.type !== "<class 'module'>"
+            ) || [];
+            setFilteredOptions(filtered);
+            setDropdownVisible(filtered.length > 0);
+            setSelectedIndex(0);
+        } else {
+            setDropdownVisible(false);
+            setFilteredOptions([]);
+        }
+    };
+
+    const handleOptionSelect = (variableName: string, parentDf?: string) => {
```

Comment:
There's a bug here where if you 

1. Have a message like `Hello`
2. Move the cursor to the start of the input, right before the H
3. Type @ then the dropdown appears
4. Select something from the dropdown
5. It inserts the new item and drops the word Hello

https://github.com/user-attachments/assets/42fcb9cf-f6a4-4ff9-82f2-69439e01d5fa